src/main/java/com/powsybl/openloadflow/OpenLoadFlowProvider.java

```diff
@@ -127,11 +128,11 @@ private LoadFlowResult runAc(Network network, LoadFlowParameters parameters, Ope
             results = AcloadFlowEngine.run(network, new LfNetworkLoaderImpl(), acParameters, reporter);
         }
 
-        boolean ok = results.stream().anyMatch(result -> result.getNewtonRaphsonStatus() == NewtonRaphsonStatus.CONVERGED);
+        boolean atLeastOneComponentHasToBeUpdated = results.stream().anyMatch(result -> result.isOk() && result.getNewtonRaphsonIterations() > 0);
 
         // do not reset state in case all results are ok and no NR iterations because it means that network was just
```

Comment:
You must put this comment to explain how you deduce that one component must be updated.